src/lib/subtract/index.ts

```diff
@@ -0,0 +1,102 @@
+import {z} from 'zod';
+
+import {ERRORS} from '../../util/errors';
+import {buildErrorMessage} from '../../util/helpers';
+import {validate} from '../../util/validations';
+
+import {PluginInterface} from '../../interfaces';
+import {PluginParams} from '../../types/common';
+import {SubtractConfig} from './types';
+
+const {InputValidationError} = ERRORS;
+
+export const Subtract = (globalConfig: SubtractConfig): PluginInterface => {
+  const errorBuilder = buildErrorMessage(Subtract.name);
+  const metadata = {
+    kind: 'execute',
+  };
+
+  /**
+   * Checks global config value are valid.
+   */
+  const validateGlobalConfig = () => {
+    const globalConfigSchema = z.object({
+      'input-parameters': z.array(z.string()),
+      'output-parameter': z.string().min(1),
+    });
+
+    return validate<z.infer<typeof globalConfigSchema>>(
+      globalConfigSchema,
+      globalConfig
+    );
+  };
+
+  /**
+   * Checks for required fields in input.
+   */
+  const validateSingleInput = (
+    input: PluginParams,
+    inputParameters: string[]
+  ) => {
+    inputParameters.forEach(metricToSubtract => {
+      validateParamExists(input, metricToSubtract);
+      validateNumericString(input[metricToSubtract]);
+    });
+
+    return input;
+  };
+
+  const validateParamExists = (input: PluginParams, param: string) => {
+    if (!input[param]) {
+      throw new InputValidationError(
+        errorBuilder({
+          message: `${param} is missing from the input array`,
+        })
+      );
+    }
+  };
+
+  const validateNumericString = (str: string) => {
+    if (isNaN(+Number(str))) {
+      throw new InputValidationError(
+        errorBuilder({
+          message: `${str} is not numberic`,
+        })
+      );
+    }
+  };
+
+  /**
+   * Subtract items from inputParams[1..n] from inputParams[0] and write the result in a new param outputParam.
+   */
+  const execute = async (inputs: PluginParams[]): Promise<PluginParams[]> => {
+    const {
+      'input-parameters': inputParameters,
+      'output-parameter': outputParameter,
+    } = validateGlobalConfig();
+    return inputs.map(input => {
+      const safeInput = validateSingleInput(input, inputParameters);
```

Comment:
you just can call `validateSingleInput(input, inputParameters);`, there isn't a necessity to assign it to the `safeInput`.

Reply:
same as i response here: https://github.com/Green-Software-Foundation/if-plugins/pull/91/files#r1601146991.

Reply:
fixed per conclusion in the referenced PR. Thanks for highlighting this :)